content/articles/2017-12-workflow-vs-script.md

```diff
@@ -0,0 +1,126 @@
+
+This use of `setwd()` is also highly suggestive that the useR does all of their work in one R process and manually switches gears when they shift from one project to another. That sort of workflow makes it unpleasant to work on more than one project at a time and also makes it easy for work done on one project to accidentially leak into subsequent work on another (e.g., objects, loaded packages, session options).
+
+## Use the [here package](https://CRAN.R-project.org/package=here) and Projects
+
+How can you avoid `setwd()` at the top of every script?
+
+  * Organize each logical project into a folder on your computer.
+  * Make sure the top-level folder advertises itself as such: I HOLD A PROJECT! This can be as simple as having an empty file named `.here`. Or, if you use RStudio and/or Git, those both leave characteristic files behind that will get the job done.
+  * Use the `here()` function from the [here package](https://CRAN.R-project.org/package=here) to build the path when you read or write a file. Create paths relative to the top-level directory.
+  
+Whenever you work on this project, launch the R process from the project's top-level directory. To continue our example, make sure R is launched in the `foofy` directory, wherever that may be. Now the code looks like so:
+
+``` r
+df <- read.delim(here("data", "raw_foofy_data.csv"))
+p <- ggplot(df, aes(x, y)) + geom_point()
+ggsave(here("figs", "foofy_scatterplot.png"))
+```
+
+This will run, with NO HAND EDITS, for anyone who follows the convention about launching R in the project folder. (In fact, it will even work if you launch R from any subfolder inside the project.) It's like agreeing that we will all drive on the left or the right. A hallmark of civilization is following conventions that constrain your behavior a little, in the name public safety.
+
+If you do use RStudio, this can be even easier. RStudio allows you to designate a folder as an [RStudio Project](https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects) (note the capital "P"). All this means is that RStudio will leave a file, e.g., `foofy.Rproj`, in the folder. Consider this an RStudio "note to self" that can hold preferences specific to that project.
+
+The here package piggybacks on the Project setup and uses the presence of an `.Rproj` file to recognize the top-level of your project. Last but not least, you can double click on the `.Rproj` file to launch an instance of RStudio where the associated R process and file browser are conveniently pointed at the correct folder. RStudio fully supports Project-based workflows, making it easy to switch from one to another, have many open at once, re-launch recently used Projects, etc.
```

Comment:
Maybe screenshot of app switcher showing a bunch open here?  I think you made a good point about this in your talk - something about easily context switching across projects within a day.

Reply:
I didn't add any inline images (yet) because reading through related PR and looking at other posts ... it wasn't immediately clear to me how to proceed. I will add later.

Reply:
@batpigandme figured it out, so will have some advice

Reply:
@jennybc in the great tradition of doing things the way someone else did them, I ended up copying what Lucy did for her post and creating a subfolder in [static/images](https://github.com/tidyverse/tidyverse.org/tree/master/static/images) and pointing to it w/ html. 😕

(I've been getting _very_ acquainted with the innards of Hugo's relative paths, and partials, so better solution to come).